Sources/MongoSwift/MongoCollection.swift

```diff
@@ -103,6 +122,41 @@ public struct MongoCollection<T: Codable> {
         return self._client.operationExecutor.execute(operation, client: self._client, session: session)
     }
 
+    /**
+     * Renames this collection with the specified options.
+     *
+     * - Parameters:
+     *   - to: A `String`, the new name for the collection
+     *   - dropTarget: A `Bool`, indicating whether the target collection should be dropped prior to renaming it.
+     *                 The default value is false.
+     *   - options: Optional `RenamedCollectionOptions` to use for the collection
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Returns:
+     *    An `EventLoopFuture<MongoCollection>` evaluating to a copy of the target `MongoCollection` with the new name.
+     *
+     *    If the future evaluates to an error, it is likely one of the following:
+     *    - `MongoError.CommandError` if an error occurs that prevents the command from executing.
+     *    - `MongoError.InvalidArgumentError` if the options passed in form an invalid combination.
+     *    - `MongoError.LogicError` if the provided session is inactive.
+     *    - `MongoError.LogicError` if this databases's parent client has already been closed.
+     *    - `EncodingError` if an error occurs while encoding the options to BSON.
+     */
+    public func renamed(
+        _ to: String,
```

Comment:
I think we want to use this label, since it makes the method invocation slightly more readable:
```swift
coll.renamed(to: "new_name")
// vs
coll.renamed("new_name")
```